dist/js/multiselect.js

```diff
@@ -1,755 +1,1812 @@
+             * @returns {number} - the initial position of this element relative to the parent element
+             */
+            function getInitialPosition($optionOrOptgroup) {
+                if ($optionOrOptgroup !== undefined && $optionOrOptgroup.is("option,optgroup")) {
+                    return $optionOrOptgroup.data(DATA_POSITION);
+                }
+                return -1;
+            }
+
+            /**
+             * Stores the initial position (index) inside the multiselect
+             * for the given option or optgroup
+             * @param {jQuery} $optionOrOptgroup - the option or optgroup for which the position should be recorded
+             * @param {number} newPosition - the new stored position for this element
+             * @returns {jQuery} - returns the element for chaining
+             */
+            function setInitialPosition($optionOrOptgroup, newPosition) {
+                if ($optionOrOptgroup instanceof jQuery && $optionOrOptgroup.is("option,optgroup")) {
+                    $optionOrOptgroup.data(DATA_POSITION, newPosition);
+                }
+                return $optionOrOptgroup;
+            }
+
+            /**
+             * Either prepends the given search filter html to the target select element
+             * just uses an existing input for the filter. Also stores which
+             * selects should be affected by the filter.
+             * @param {jQuery} $targetSelect - the select element where the filter should be prepended
+             * @param {string|jQuery} searchFilterHtmlOrElement - the html for the search filter or an existing jQuery element
+             * @param {jQuery} $filteredSelects - the select elements that should be filtered with this input
+             * @returns {FilterRelation} - the filter/selects relation
+             */
+            function prependSearchFilter($targetSelect, searchFilterHtmlOrElement, $filteredSelects) {
+                /** The created search filter element
+                 * @type {jQuery} */
+                var $searchFilter = $();
+                if (typeof searchFilterHtmlOrElement === "string") {
+                    $searchFilter = $(searchFilterHtmlOrElement);
+                    $targetSelect.before($searchFilter);
+                } else if (searchFilterHtmlOrElement instanceof $) {
+                    //noinspection JSValidateTypes
+                    $searchFilter = searchFilterHtmlOrElement;
+                }
+                return {
+                    $filterInput: $searchFilter,
+                    $filteredSelects: $filteredSelects
+                };
+            }
+
+            /**
+             * The old filter function stored for reference.
+             * @param {string} filterValue - the value options must match
+             * @param {jQuery} $select - the select to filter
+             */
+            function oldFilterOptions(filterValue, $select) {
```

Comment:
This is basically your old filter function, which I kept for performance measuring, because I was unsure. It seems mine is faster, but that could be incidental.